pkg/lib/oauth/handler/handler_anonymous_user.go

```diff
@@ -165,16 +165,24 @@ func (h *AnonymousUserHandler) signupAnonymousUserWithRefreshTokenSessionType(
 			return nil, ErrLoggedInAsNormalUser
 		}
 
+		session, ok := grant.ToSession(currentRefreshTokenHash)
+		if !ok {
+			panic(fmt.Errorf("unexpected: failed to resolve offline grant session"))
+		}
+
 		resp := protocol.TokenResponse{}
 		issueAccessGrantOptions := oauth.IssueAccessGrantOptions{
-			ClientConfig:       client,
-			Scopes:             scopes,
-			AuthorizationID:    authz.ID,
-			AuthenticationInfo: grant.GetAuthenticationInfo(),
-			SessionLike:        grant,
-			RefreshTokenHash:   refreshTokenHash,
+			ClientConfig:            client,
+			Scopes:                  scopes,
+			AuthorizationID:         authz.ID,
+			AuthenticationInfo:      grant.GetAuthenticationInfo(),
+			SessionLike:             grant,
+			InitialRefreshTokenHash: session.InitialTokenHash,
 		}
-		err = h.TokenService.IssueAccessGrant(ctx, issueAccessGrantOptions, resp)
+		err = h.TokenService.IssueAccessGrantByRefreshToken(ctx, IssueAccessGrantByRefreshTokenOptions{
+			IssueAccessGrantOptions:  issueAccessGrantOptions,
+			ShouldRotateRefreshToken: client.RefreshTokenRotationEnabled,
```

Comment:
This should be false. We do not rotate refresh tokens in anonymous user.

Change to

```golang
ShouldRotateRefreshToken: false, // We do not rotate refresh tokens in anonymous user.
```